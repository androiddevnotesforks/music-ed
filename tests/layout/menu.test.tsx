import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom"
import Menu from "../../components/layout/menu";

const mockProps = {
  userId: "Mock User ID"
}

describe("Menu component", () => {
  beforeEach(() => {
    render(<Menu {...mockProps}/>)
  })
  it("Renders", () => {
    const menu = screen.getByTestId("menu-div")
    expect(menu).toBeInTheDocument()
  })
  it("Matches snapshot", () => {
    const menu = screen.getByTestId("menu-div")
    expect(menu).toMatchSnapshot()
  })
  it("About link exists", () => {
    const menu = screen.getByTestId("menu-div")
    expect(menu.textContent).toMatch("About")
  })
  it("Exercises link exists", () => {
    const menu = screen.getByTestId("menu-div")
    expect(menu.textContent).toMatch("Exercises")
  })
  it("My Progress link exists", () => {
    const menu = screen.getByTestId("menu-div")
    expect(menu.textContent).toMatch("My Progress")
  })
  it("Contact link exists", () => {
    const menu = screen.getByTestId("menu-div")
    expect(menu.textContent).toMatch("Contact")
  })
})