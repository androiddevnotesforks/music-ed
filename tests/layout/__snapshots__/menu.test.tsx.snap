// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component Matches snapshot 1`] = `
<div
  class="menu"
  data-testid="menu-div"
>
  <div
    class="menu-list"
  >
    <a
      href="/about"
    >
      <h3
        class="menu-items"
      >
        About
      </h3>
    </a>
    <a
      href="/exercises"
    >
      <h3
        class="menu-items"
      >
        Exercises
      </h3>
    </a>
    <a
      href="/user/Mock User ID"
    >
      <h3
        class="menu-items"
      >
        My Progress
      </h3>
    </a>
    <a
      href="/contact"
    >
      <h3
        class="menu-items"
      >
        Contact
      </h3>
    </a>
  </div>
</div>
`;
